acasclient/acasclient.py

```diff
@@ -290,24 +313,51 @@ def parse_sdf(file_content):
 
 class client():
 
-    def __init__(self, creds):
+    def __init__(self, creds, enable_retries=True, retry_config: Retry=None):
+        """ACAS Client
+        Initializes the ACAS client.
+        
+        Args:
+            creds: A dict of credentials with the following keys
+                username: ACAS username
+                password: ACAS password
+                url: ACAS server URL
+            enable_retries: Boolean to enable retry session
+            retry_config: Optional requests.adapters.Retry object to configure retry behavior
+            
+            Returns:
+            An ACAS client object.
+        """
         self.username = creds['username']
         self.password = creds['password']
         self.url = creds['url']
-        self.session = self.getSession()
+        self.session = self._create_session(retry=enable_retries, retry_config=retry_config or DEFAULT_RETRY_CONFIG)
+        self.idempotent_post_session = self._create_session(retry=enable_retries, retry_config=retry_config or DEFAULT_IDEMPOTENT_POST_RETRY_CONFIG)
```

Comment:
This second session is purely for the retryable (i.e. idempotent) POST methods. If retries are disabled, it'll behave the same as the default session. Otherwise it will retry even POSTs, so it is only referenced for API routes that are safe to retry.

Reply:
Yep, makes sense. This led me to lightly refactor `_create_session` into smaller functions so we can mix & match them. Now I'm creating two sessions but only logging in once.